L1Trigger/L1TGlobal/src/AXOL1TLCondition.cc

```diff
@@ -0,0 +1,254 @@
+
+// system include files
+#include <iostream>
+#include <iomanip>
+
+#include <string>
+#include <vector>
+#include <algorithm>
+#include "ap_fixed.h"
+#include "hls4ml/emulator.h"
+
+// user include files
+//   base classes
+#include "L1Trigger/L1TGlobal/interface/AXOL1TLTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/ConditionEvaluation.h"
+
+#include "L1Trigger/L1TGlobal/interface/MuCondition.h"
+#include "L1Trigger/L1TGlobal/interface/AXOL1TLCondition.h"
+#include "L1Trigger/L1TGlobal/interface/CaloCondition.h"
+#include "L1Trigger/L1TGlobal/interface/EnergySumCondition.h"
+#include "L1Trigger/L1TGlobal/interface/MuonTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/CaloTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/EnergySumTemplate.h"
+#include "L1Trigger/L1TGlobal/interface/GlobalScales.h"
+
+#include "DataFormats/L1Trigger/interface/L1Candidate.h"
+
+#include "L1Trigger/L1TGlobal/interface/GlobalBoard.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/MessageLogger/interface/MessageDrop.h"
+
+// constructors
+//     default
+l1t::AXOL1TLCondition::AXOL1TLCondition() : ConditionEvaluation() {
+  // empty
+}
+
+//     from base template condition (from event setup usually)
+l1t::AXOL1TLCondition::AXOL1TLCondition(const GlobalCondition* axol1tlTemplate, const GlobalBoard* ptrGTB)
+    : ConditionEvaluation(),
+      m_gtAXOL1TLTemplate(static_cast<const AXOL1TLTemplate*>(axol1tlTemplate)),
+      m_gtGTB(ptrGTB) {}
+
+// copy constructor
+void l1t::AXOL1TLCondition::copy(const l1t::AXOL1TLCondition& cp) {
+  m_gtAXOL1TLTemplate = cp.gtAXOL1TLTemplate();
+  m_gtGTB = cp.gtGTB();
+
+  m_condMaxNumberObjects = cp.condMaxNumberObjects();
+  m_condLastResult = cp.condLastResult();
+  m_combinationsInCond = cp.getCombinationsInCond();
+
+  m_verbosity = cp.m_verbosity;
+}
+
+l1t::AXOL1TLCondition::AXOL1TLCondition(const l1t::AXOL1TLCondition& cp) : ConditionEvaluation() { copy(cp); }
+
+// destructor
+l1t::AXOL1TLCondition::~AXOL1TLCondition() {
+  // empty
+}
+
+// equal operator
+l1t::AXOL1TLCondition& l1t::AXOL1TLCondition::operator=(const l1t::AXOL1TLCondition& cp) {
+  copy(cp);
+  return *this;
+}
+
+// methods
+void l1t::AXOL1TLCondition::setGtAXOL1TLTemplate(const AXOL1TLTemplate* caloTempl) { m_gtAXOL1TLTemplate = caloTempl; }
+
+///   set the pointer to uGT GlobalBoard
+void l1t::AXOL1TLCondition::setuGtB(const GlobalBoard* ptrGTB) { m_gtGTB = ptrGTB; }
+
+const bool l1t::AXOL1TLCondition::evaluateCondition(const int bxEval) const {
+  bool condResult = false;
+  int useBx = bxEval + m_gtAXOL1TLTemplate->condRelativeBx();
+  cout << "Considering BX " << useBx << std::endl;
+
+  //HLS4ML stuff
+  std::string AXOL1TLmodelversion = "GTADModel_v1";
+  cout << "loading model... " << AXOL1TLmodelversion << std::endl;
+  hls4mlEmulator::ModelLoader loader(AXOL1TLmodelversion);
+  std::shared_ptr<hls4mlEmulator::Model> model;
+  model = loader.load_model();
+  cout << "model loaded! " << std::endl;
```

Comment:
The AXOL1TL model version should probably not be hard-coded into the emulator. This means that at any point if the emulator needs a new AXOL1TL model, the emulator needs to be updated, and then a new release made. This should be configurable as part of the emulator configuration options.

Also please change the `cout`'s to `LogInfo` or remove them

Reply:
I agree that it would be better to have the model version more flexible. Do you have any suggestions for how to better configure it?

Reply:
This will likely have to be configured from the uGT overall `_cfi`. The only input needed here is a string, so passing in a `cms.untracked.string` (or tracked, whatever), from the uGT python configuration is the shorted way to do this.